### PR TITLE
[harness-simulation] change config format from JSON to YAML

### PR DESCRIPTION
**Pull request by jcdong98**
This commit changes the format of the configuration file from JSON to YAML, as YAML is designed for human interaction and better suitable for configuration than JSON.

### CODE REVIEW

**Review by superwhd — APPROVED** [review]
LGTM:+1:

File: tools/harness-simulation/posix/install.sh [hunk 1]
```diff
@@ -55,6 +47,12 @@ sudo apt install -y "${PACKAGES[@]}"
 pip3 install -r "${POSIX_DIR}/requirements.txt"
 python3 -m grpc_tools.protoc -I"${SNIFFER_DIR}" --python_out="${SNIFFER_DIR}" --grpc_python_out="${SNIFFER_DIR}" proto/sniffer.proto
 
+CONFIG_NAME=${1:-"${POSIX_DIR}/config.yml"}
+# convert YAML to JSON
+CONFIG=$(python3 -c 'import json, sys, yaml; print(json.dumps(yaml.safe_load(open(sys.argv[1]))))' "$CONFIG_NAME")
```
Comment (simonlingoogle) [inline tools/harness-simulation/posix/install.sh:52]: better if we can install `yq`?

Comment (jcdong98) [reply tools/harness-simulation/posix/install.sh:52]: It will be much inconvenient that `yq` cannot be installed by the default package manager like `apt`.

One approach mentioned in README of https://github.com/mikefarah/yq is to install `yq` by `snap`, but `snap` should be installed first. Also, on my Raspberry Pi, `yq` is located in `/snap/bin/yq`, but `/snap/bin` is not in `$PATH`.

Another feasible approach is to download `yq` binary directly. However, we have to specify the version and the architecture (amd64, arm64, etc). The architecture may differ among different devices. For example, `uname -r` may return `aarch64` or `armv7l` on different Raspberry Pi models.

In conclusion, I think converting YAML to JSON first and then using `jq` to process in Bash should be the easiest way.

Comment (simonlingoogle) [reply tools/harness-simulation/posix/install.sh:52]: sgtm.

**Review by simonlingoogle — APPROVED** [review]
LGTM 👍

**Review by jwhui — APPROVED** [review]
